### PR TITLE
perf: reduce selected-token logprob memory

### PR DESCRIPTION
**Pull request by mohamed-alansary**
## Summary

This PR reduces peak memory in AutoModel local logprob computation when `logprob_chunk_size` is configured.

Instead of computing and storing full-vocab log-probability chunks, concatenating them back into a `[batch, sequence, vocab]` tensor, and only then gathering selected next-token logprobs, the chunked path now gathers selected token logprobs inside each chunk and concatenates only the compact `[batch, chunk]` results.

## Motivation

For long-context runs with large vocabularies, the previous chunked implementation could still materialize very large fp32 tensors during local logprob computation. This defeats much of the benefit of chunking because `torch.cat(chunked_log_probs, dim=1)` reconstructs the full `[batch, sequence, vocab]` logprob tensor before gathering the selected tokens.

The downstream output only needs one logprob per next token, so keeping full-vocab logprob chunks alive is unnecessary.

## Changes

- Reuses the existing `logprob_chunk_size` setting.
- Slices logits to the next-token target sequence length before logprob computation.
- Computes `log_softmax` per sequence chunk.
- Gathers selected next-token logprobs inside each chunk.
- Concatenates only selected-token `[batch, chunk]` tensors.
- Preserves fp32 log-softmax behavior.
- Preserves top-k/top-p filtering behavior.
- Handles empty next-token sequences by returning an empty fp32 tensor.

## Testing

- `ruff check --fix nemo_rl/models/automodel/train.py tests/unit/models/automodel/test_logprob_memory.py`
- `python -m pytest tests/unit/models/automodel/test_logprob_memory.py -v`
  - `9 passed`

### CODE REVIEW

**Review by yuki-97 — DISMISSED** [review]
LGTM, thanks for catching and fixing this!

**Comment by yuki-97** [comment]
hi @mohamed-alansary , there's a lint check fail, could you help to fix with `pre-commit run --all-files`?

**Comment by mohamed-alansary** [comment]
Thanks @yuki-97 ! Fixed the lint/style issue. Could you please re-review/re-trigger CI?

**Review by yuki-97 — APPROVED** [review]
